1.Core/UIElements/Drivers/BrowserstackWebDriver.cs

```diff
@@ -0,0 +1,102 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Edge;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.Remote;
+using OpenQA.Selenium.Safari;
+using Todoly.Core.Helpers;
+
+namespace Todoly.Core.UIElements.Drivers;
+
+internal class BrowserStackOptions : DriverOptions
+{
+    public BrowserStackOptions(string browser_name, string browser_version)
+    {
+        BrowserName = browser_name;
+        BrowserVersion = browser_version;
+    }
+
+    [Obsolete]
+    public override void AddAdditionalCapability(string capabilityName, object capabilityValue)
+    {
+        AddAdditionalOption(capabilityName, capabilityValue);
+    }
+
+    public override ICapabilities ToCapabilities()
+    {
+        IWritableCapabilities capabilities = GenerateDesiredCapabilities(true);
+        return capabilities.AsReadOnly();
+    }
+}
+
+public class BrowserstackWebDriverFactory
+{
+    public static readonly string BROWSERSTACK_USERNAME = ConfigModel.BROWSERSTACK_USERNAME;
+    public static readonly string BROWSERSTACK_ACCESSKEY = ConfigModel.BROWSERSTACK_ACCESSKEY;
+
+    public static IWebDriver GetDriver()
+    {
+        string driverType = ConfigBuilder.Instance.GetString("ui", "DriverType");
+        switch (driverType)
+        {
+            case "Chrome":
+                var chrome = new RemoteWebDriver(
+                    new Uri("https://hub.browserstack.com/wd/hub/"),
```

Comment:
If you need to pass the BrowserStack url to the remote driver, it shouldn't be here, and it should be stored on an environment variable as well.